### PR TITLE
IDEMPIERE-5909:use jdbc library as bundle

### PR DESCRIPTION
**Pull request by hieplq**
# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [ ] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
### Documentation
- [ ] I have made corresponding changes to the documentation as follows:
- - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
- - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.compiere.db.oracle.provider/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -6,10 +6,10 @@ Bundle-Version: 11.0.0.qualifier
 Bundle-RequiredExecutionEnvironment: JavaSE-17
 Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version>=17))"
 Require-Bundle: org.adempiere.base;bundle-version="0.0.0",
- org.adempiere.install;bundle-version="0.0.0";resolution:=optional
-Bundle-ClassPath: .,
- lib/HikariCP.jar,
- lib/ojdbc10.jar
+ org.adempiere.install;bundle-version="0.0.0";resolution:=optional,
+ com.zaxxer.HikariCP;bundle-version="5.0.1",
+ wrapped.com.oracle.database.jdbc.ojdbc10;bundle-version="19.16.0"
```
Comment (CarlosRuiz-globalqss) [inline org.compiere.db.oracle.provider/META-INF/MANIFEST.MF:11]: latest ojdbc10 is 19.21.0

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]

**Comment by CarlosRuiz-globalqss** [comment]
Hi @hieplq 

I think when changing the targetplatform is used to change the target name
org.idempiere.p2.targetplatform/org.idempiere.p2.targetplatform.target
maybe to something like
<target name="idempiere-231107"
I don't know if the sequenceNumber must be changed too -  @hengsin?


Now, I ran the unit tests in postgresql successfully.

But oracle is showing 5 errors and 3 failures:

1 - at org.idempiere.test.model.PurchaseOrderTest.testQtyReservedForNegativeOrderAndReceipt(PurchaseOrderTest.java:116)
org.adempiere.exceptions.AdempiereException: SaveError

2 - at org.idempiere.test.model.ProductionTest.testMultipleASI(ProductionTest.java:336)
org.adempiere.exceptions.AdempiereException: ORA-00060: Deadlock beim Warten auf Ressource festgestellt

3 - at org.idempiere.test.model.ProductionTest.testMultipleDateMPolicy(ProductionTest.java:445)
org.adempiere.exceptions.AdempiereException: DeleteError

4 - at org.idempiere.test.model.MTaxTest.testDistributeTaxToProductCost(MTaxTest.java:253)
org.adempiere.exceptions.AdempiereException: ORA-00060: Deadlock beim Warten auf Ressource festgestellt

5 - at org.idempiere.test.model.BankStatementTest.testReversalOfReconciledPayment1(BankStatementTest.java:136)
org.adempiere.exceptions.AdempiereException: Error: : Database Error.: DBExecuteError:ORA-00060: Deadlock beim Warten auf Ressource festgestellt


Failures:

1 - at org.idempiere.test.form.MergeFormTest.testMergeUser(MergeFormTest.java:254)
org.opentest4j.AssertionFailedError: Merge fail ==> expected: <true> but was: <false>

2 - at org.idempiere.test.base.MatchInvTest.testMatReceiptPosting(MatchInvTest.java:344)
org.opentest4j.AssertionFailedError: MatchInv incorrect amount posted 0 ==> expected: <36.00> but was: <30.60>

3 - at org.idempiere.test.base.MatchInvTest.testCreditMemoPosting(MatchInvTest.java:462)
org.opentest4j.AssertionFailedError: MatchInv incorrect amount posted 324 ==> expected: <324.00> but was: <275.40>


NOTE: I haven't tested the installers

**Comment by CarlosRuiz-globalqss** [comment]
@hengsin confirmed that is required to change the sequence number too:
> the minimum require is to increment the sequence number by 1

**Comment by CarlosRuiz-globalqss** [comment]
I got corrected from @hengsin:

> you can keep the target name unchange and increase the sequence number by 1.
> If you does change the target name, it doesn't really matter whether you keep or change the sequence number 

My preference is to change the name to reflect the date of the target platform update.

**Comment by hengsin** [comment]
note that if you update the target name, please update the target name in utils_dev/oomph/idempiere.setup too.

**Comment by hieplq** [comment]
i also run test case before and just run it again. i get only bellow issue, that why i think it's not relate my change. wait me to check it on

image

> Hi @hieplq
> 
> I think when changing the targetplatform is used to change the target name org.idempiere.p2.targetplatform/org.idempiere.p2.targetplatform.target maybe to something like <target name="idempiere-231107" I don't know if the sequenceNumber must be changed too - @hengsin?
> 
> Now, I ran the unit tests in postgresql successfully.
> 
> But oracle is showing 5 errors and 3 failures:
> 
> 1 - at org.idempiere.test.model.PurchaseOrderTest.testQtyReservedForNegativeOrderAndReceipt(PurchaseOrderTest.java:116) org.adempiere.exceptions.AdempiereException: SaveError
> 
> 2 - at org.idempiere.test.model.ProductionTest.testMultipleASI(ProductionTest.java:336) org.adempiere.exceptions.AdempiereException: ORA-00060: Deadlock beim Warten auf Ressource festgestellt
> 
> 3 - at org.idempiere.test.model.ProductionTest.testMultipleDateMPolicy(ProductionTest.java:445) org.adempiere.exceptions.AdempiereException: DeleteError
> 
> 4 - at org.idempiere.test.model.MTaxTest.testDistributeTaxToProductCost(MTaxTest.java:253) org.adempiere.exceptions.AdempiereException: ORA-00060: Deadlock beim Warten auf Ressource festgestellt
> 
> 5 - at org.idempiere.test.model.BankStatementTest.testReversalOfReconciledPayment1(BankStatementTest.java:136) org.adempiere.exceptions.AdempiereException: Error: : Database Error.: DBExecuteError:ORA-00060: Deadlock beim Warten auf Ressource festgestellt
> 
> Failures:
> 
> 1 - at org.idempiere.test.form.MergeFormTest.testMergeUser(MergeFormTest.java:254) org.opentest4j.AssertionFailedError: Merge fail ==> expected: but was:
> 
> 2 - at org.idempiere.test.base.MatchInvTest.testMatReceiptPosting(MatchInvTest.java:344) org.opentest4j.AssertionFailedError: MatchInv incorrect amount posted 0 ==> expected: <36.00> but was: <30.60>
> 
> 3 - at org.idempiere.test.base.MatchInvTest.testCreditMemoPosting(MatchInvTest.java:462) org.opentest4j.AssertionFailedError: MatchInv incorrect amount posted 324 ==> expected: <324.00> but was: <275.40>
> 
> NOTE: I haven't tested the installers

**Comment by CarlosRuiz-globalqss** [comment]
Ah, forgot to mention, my tests were running idempiere.unit.test.parallel on oracle

Maybe the parallel creates those deadlocks

**Comment by CarlosRuiz-globalqss** [comment]
Right - running the not parallel version throws just one error:

org.adempiere.exceptions.AdempiereException: Could not save record - Require unique data:  

at org.idempiere.test.model.MTaxTest.testDistributeTaxToProductCost(MTaxTest.java:126)
Please change information.: ORA-00001: Unique Constraint (IDEMPIEREDEV.C_TAXCATEGORY_NAME) verletzt

**Comment by CarlosRuiz-globalqss** [comment]
Hi @hieplq 

Mistery solved.

I ran the unit tests on a vanilla database and 392 tests passed successfully.

The problem was caused because when the parallel tests failed left a C_TaxCategory record there, and subsequent runs fail because the name is duplicate.

So, I think this is ready to move ahead.

Can you please apply the following patch:
[pr2092PR.patch.txt](https://github.com/idempiere/idempiere/files/13283795/pr2092PR.patch.txt)

Regards,

Carlos Ruiz

**Review by CarlosRuiz-globalqss — APPROVED** [review]

**Pull request merged**